tests/shared/src/test/scala/scala/meta/tests/quasiquotes/ErrorSuite.scala

```diff
@@ -720,19 +720,21 @@ class ErrorSuite extends FunSuite {
   }
 
   test("#448") {
-    assert(typecheckError("""
+    assertNoDiff(
+      typecheckError("""
       import scala.meta._
       val notReallyAParent = t"_root_.scala.AnyVal"
-      q"class C extends $notReallyAParent"
-    """).replace("\r", "") == """
-      |<macro>:4: type mismatch when unquoting;
-      | found   : scala.meta.Type.Select
-      | required: scala.meta.Template
-      |Note: This shape of a quasiquote tells scala.meta that you're unquoting a template.
-      |If you'd like to unquote a parent reference, add {} immediately after the unquote.
-      |For more details, see https://github.com/scalameta/scalameta/issues/223.
-      |      q"class C extends $notReallyAParent"
-      |                        ^
-    """.trim.stripMargin)
+      q"class C $notReallyAParent"
```

Comment:
this type of quasiquote is probably meaningless. as a result of your [welcome] change, there's no more confusion between matching a template and a parent class.

anyone who uses `$foo` without `extends` would know it matches a template. and using `extends $foo` would now require `init"..."` and nothing else.

so i would do this:
- in `ConversionMacros.scala:typeMismatchMessage`, replace the entire template-related block (and `MetaTemplate` field) and remove the `wholeMessage` variable as well.
- change this `#448` test into two:
  - `q"class C $foo` where `foo = t"..."` (this one); wants template, gets type
  - `q"class C extends $foo` where `foo = t"..."` (original, with `extends`); wants init, gets type

Reply:
Did both, I think we should be fine now.